scripts/vehicle_node.py

```diff
@@ -0,0 +1,292 @@
+                time = rospy.Duration(0.05)
+            else:
+                time = event.current_real - event.last_real
+        
+        time = time.to_sec()
+
+        #Calculate angle turned in the given time using omega = theta/time
+        angle = velocity_data.angular.z*time
+
+        #Calculate distance travelled in the given time using linear velocity = arc distance/time
+        distance = velocity_data.linear.x*time
+
+        #Calculate yaw of the robot
+        self.vehicle_yaw += angle
+
+        #Calculate vehicle x, y, z position coordinates
+        # TODO recalculate the position based on traveling in a circular arc.
+        self.pose.position.x += (distance)*cos(self.vehicle_yaw)
+        self.pose.position.y += (distance)*sin(self.vehicle_yaw)
+
+        #Calculate z position using linear interpolation and create cloud array
+        
+        #1. Create range to be used in interpolation function
+        x = np.arange(0, self.gaussian_array.shape[0]*self.resolution, self.resolution)
+        y = np.arange(0, self.gaussian_array.shape[1]*self.resolution, self.resolution)
+
+        #2. Create cloud array to be converted to point cloud for vizualization
+        #TODO do this without for loops
+        cloud = []
+        for i in range(self.gaussian_array.shape[0]):
+            for j in range(self.gaussian_array.shape[1]):
+                innerlist = []
+                innerlist.append(x[i])
+                innerlist.append(y[j])
+                innerlist.append(self.gaussian_array[i][j])
+                cloud.append(innerlist)
+
+        #3. Create interpolation function based on the ranges and gaussian data
+        f = RectBivariateSpline(x,y, self.gaussian_array)
+
+        #4. Find z value for x and y coordinate of vehicle using interpolation function
+        self.pose.position.z = f(self.pose.position.x, self.pose.position.y)
+
+        #Convert Euler Angles to Quarternion
+        q = tf.transformations.quaternion_from_euler(0.0, 0.0, self.vehicle_yaw)
+
+        #Broadcast vehicle frame which is a child of the world frame
+        br = tf.TransformBroadcaster()
+        br.sendTransform((self.pose.position.x, self.pose.position.y, self.pose.position.z), 
+                        q, rospy.Time.now(),"vehicle_frame", "map")
+
+        #Construct the homogenous transformation matrix for map to vehicle frame
+        translation = [self.pose.position.x, self.pose.position.y, self.pose.position.z]
+        map_t_vehicle = tf.transformations.translation_matrix(translation)
+        map_R_vehicle   = tf.transformations.quaternion_matrix(q)
+        map_T_vehicle = np.zeros((4,4))
+        map_T_vehicle[0:3,0:3] = map_R_vehicle[0:3,0:3]
+        map_T_vehicle[:4,3] = map_t_vehicle[:4,3]
+
+        #Create footprint of vehicle
+        x1 = np.linspace((-self.vehicle_length), (self.vehicle_length),30)
+        y1 = np.linspace((-self.vehicle_width), (self.vehicle_width),15)
```

Comment:
Based on x1 & y1, it looks like the resulting width of the vehicle is (self.vehicle_width * 2).

Either change the variable name of `self.vehicle_width`, or change the math to match the naming.